src/NzbDrone.Core/MediaFiles/RenameMovieFileService.cs

```diff
@@ -139,38 +139,27 @@ private void RenameFiles(List<MovieFile> movieFiles, Movie movie, string oldMovi
 		public void RenameMoviePath(Movie movie, bool shouldRenameFiles = true)
 		{
 			var newFolder = _filenameBuilder.BuildMoviePath(movie);
-	        if (newFolder != movie.Path && movie.PathState == MoviePathState.Dynamic)
-	        {
-
-	            if (!_configService.AutoRenameFolders)
-	            {
-	                _logger.Info("{0}'s movie should be {1} according to your naming config.", movie, newFolder);
-	                return;
-	            }
+            if (newFolder != movie.Path && movie.PathState == MoviePathState.Dynamic)
+            {
+                if (!_configService.AutoRenameFolders)
+                {
+                    _logger.Info("{0}'s movie should be {1} according to your naming config.", movie, newFolder);
+                    return;
+                }
 
-	             _logger.Info("{0}'s movie folder changed to: {1}", movie, newFolder);
+                _logger.Info("{0}'s movie folder changed to: {1}", movie, newFolder);
                 var oldFolder = movie.Path;
                 movie.Path = newFolder;
 
-	            _diskProvider.MoveFolder(oldFolder, movie.Path);
-
-				if (false)
-				{
-					var movieFiles = _mediaFileService.GetFilesByMovie(movie.Id);
-					_logger.ProgressInfo("Renaming movie files for {0}", movie.Title);
-					RenameFiles(movieFiles, movie, oldFolder);
-					_logger.ProgressInfo("All movie files renamed for {0}", movie.Title);
-				}
-
-				_movieService.UpdateMovie(movie);
-
-                if (_diskProvider.GetFiles(oldFolder, SearchOption.AllDirectories).Count() == 0)
```

Comment:
This is needed, else you will have leftover folders with no files in it. If you want to fix the exception, either wrap it in a try catch block or check if the folder exists beforehand.

Reply:
I might be missing something, but on line 155 the old folder get moved to the new path. After that there should be no old folder to cleanup.

All of this happen in the same code path.

Reply:
Any hints?